components-api/src/main/scala/pl/touk/nussknacker/engine/api/Params.scala

```diff
@@ -23,7 +23,7 @@ final case class Params(nameToValueMap: Map[String, Any]) {
       .getOrElse(throw new IllegalArgumentException(cannotFindParamNameMessage(paramName)))
   }
 
-  private def rawValueExtract(paramName: String) = nameToValueMap.get(paramName).flatMap(Option.apply)
+  private def rawValueExtract(paramName: String) = nameToValueMap.get(paramName)
```

Comment:
AFAIR, it is needed by the database lookup component. That's why the tests failed

Reply:
But with this `Option` wrap, user can get error like this:
`Cannot find param name [foo]. Available param names: bar,foo,xyz`
when `foo` exists in `nameToValueMap` with `null` value.
Which is a little bit strange.

Reply:
SPeL is "null oriented", user can provide an expression `null`. So we have two optional things in this equation:
- parameter is missing - probably because someone import some outdated scenario version which wasn't migrated or developer didn't provide the migration for new changes in the component
- parameter is defined, and the expression has eagerly known, `null` value.

Currenlty we flatten both things which can generate confusions. If db lookup component base on this flattening, IMO we should fix it. It would be nice to provide a test that show both situations. We have a lot of combinations in  `ProcessTestHelpers`

Reply:
done